mount/mount.go

```diff
@@ -76,6 +77,27 @@ type MountPoint struct {
 	Pass   int
 }
 
+type MountError struct {
+	FilesystemMismatch  bool
+	HasFileSystemErrors bool
+	UnformattedReadOnly bool
+	FormatFailed        bool
+	Message             string
+}
+
+func (mountError *MountError) String() string {
+	return mountError.Message
+}
+
+func (mountError *MountError) Error() string {
+	return mountError.Message
+}
+
+func (mountError *MountError) withMessage(format string, args ...interface{}) *MountError {
```

Comment:
I'd prefer `NewMountError(reason <some enum>, message string)`. But it depends on existence of such enum, see above.